src/journal_entry.rs

```diff
@@ -27,20 +29,32 @@ impl JournalEntry {
 		Self::from(fields)
 	}
 
-	pub fn get_field(&self, field: &str) -> Option<&str> {
-		self.fields.get(field).map(|v| v.as_str())
+	pub fn get_field_binary(&self, field: &str) -> Option<&[u8]> {
+		self.fields.get(field).map(|v| v.as_slice())
+	}
+
+	pub fn get_field_string(&self, field: &str) -> Result<Option<&str>, str::Utf8Error> {
+		self.fields
+			.get(field)
+			.map(|v| str::from_utf8(v))
+			.transpose()
+	}
+
+	pub fn get_field_string_lossy(&self, field: &str) -> Option<Cow<'_, str>> {
+		self.fields.get(field).map(|v| String::from_utf8_lossy(v))
 	}
 
 	pub fn get_fields(&self) -> &JournalEntryFields {
 		&self.fields
 	}
 
-	pub fn get_message(&self) -> Option<&str> {
-		self.fields.get("MESSAGE").map(|v| v.as_ref())
+	pub fn get_message(&self) -> Option<Cow<'_, str>> {
+		self.get_field_string_lossy("MESSAGE")
```

Comment:
I'm not a huge fan of being lossy by default but up to you. It is supposed to be convenient so i get the desire.

Maybe at least add a warning to the docs?